### PR TITLE
Problem (Fix #987): byzantine punishment doesn’t take evidence age in to account

### LINKED ISSUE #987: Problem: byzantine punishment doesn't take evidence age into account
https://docs.tendermint.com/master/spec/abci/abci.html#evidenceparams

" Evidence older than this is considered stale and ignored."

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- Fetch max evidence age from consensus parameter and store in chain
state
- Filter evidences against it

### CODE REVIEW

File: test-common/src/chain_env.rs [hunk 1]
```diff
@@ -321,6 +321,19 @@ impl ChainEnv {
             app_state_bytes: serde_json::to_vec(&self.init_config).unwrap(),
             chain_id: TEST_CHAIN_ID.to_owned(),
             validators,
+            consensus_params: Some(ConsensusParams {
+                evidence: Some(EvidenceParams {
+                    max_age_duration: Some(::protobuf::well_known_types::Duration {
+                        seconds: 172_800,
```
Comment (leejw51crypto) [inline test-common/src/chain_env.rs:327]: some explanation of this number
how about const variable?

Comment (yihuang) [reply test-common/src/chain_env.rs:327]: It's just 48 hours in seconds, the default config value of tendermint.
Only used in unit tests.

File: chain-abci/tests/punishment.rs [hunk 2]
```diff
@@ -70,6 +71,34 @@ fn begin_block_should_jail_byzantine_validators() {
...
+    let mut app = env.chain_node(storage);
+    let _rsp_init_chain = app.init_chain(&env.req_init_chain());
+
+    // Begin Block
+    let mut req = RequestBeginBlock {
+        byzantine_validators: vec![env.byzantine_evidence(0)].into(),
+        ..env.req_begin_block(1, 0)
+    };
+    req.header.as_mut().unwrap().time = Some(Timestamp {
+        seconds: 172801,
```
Comment (leejw51crypto) [inline chain-abci/tests/punishment.rs:87]: const MAX_EVIDENCE_VALUE= 172_800
MAX_EVIDENCE_VALUE+1?

Comment (yihuang) [reply chain-abci/tests/punishment.rs:87]: Changed to fetch the value from `app`.

File: chain-abci/src/app/mod.rs [hunk 3]
```diff
@@ -126,7 +126,7 @@ impl<T: EnclaveProxy> abci::Application for ChainNodeApp<T> {
         };
 
         // ignore the invalid items (logged)
-        let evidences = req
+        let mut evidences = req
```
[the remark this answers is not in the document]
Comment (yihuang) [reply chain-abci/src/app/mod.rs]: Thanks, done

**Pull request merged**